libnavigator/src/main/java/com/mapbox/navigation/navigator/MapboxNativeNavigatorImpl.kt

```diff
@@ -120,6 +157,172 @@ object MapboxNativeNavigatorImpl : MapboxNativeNavigator {
     }
 
     private fun NavigationStatus.getRouteProgress(): RouteProgress {
-        return RouteProgress("")
+        val upcomingStepIndex = stepIndex + ONE_INDEX
+
+        val routeProgressBuilder = RouteProgress.Builder()
+        val legProgressBuilder = RouteLegProgress.Builder()
+        val stepProgressBuilder = RouteStepProgress.Builder()
+
+        ifNonNull(route) { route ->
+            ifNonNull(route.legs()) { legs ->
```

Comment:
we can join this two checks with one
```suggestion
            ifNonNull(route?.legs()) { legs ->
```